internal/resource/config/oauth/oauth_auth_server_settings_resource.go

```diff
@@ -0,0 +1,1167 @@
+	}
+
+	// Read the response into the state
+	readOauthAuthServerSettingsResponse(ctx, apiReadOauthAuthServerSettings, &state)
+
+	// Set refreshed state
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *oauthAuthServerSettingsResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	updateOauthAuthServerSettings(ctx, req, resp, r.apiClient, r.providerConfig)
+}
+
+func updateOauthAuthServerSettings(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse, apiClient *client.APIClient, providerConfig internaltypes.ProviderConfiguration) {
+	// Retrieve values from plan
+	var plan oauthAuthServerSettingsResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get the current state to see how any attributes are changing
+	var state oauthAuthServerSettingsResourceModel
+	req.State.Get(ctx, &state)
+	updateOauthAuthServerSettings := apiClient.OauthAuthServerSettingsApi.UpdateAuthorizationServerSettings(config.ProviderBasicAuthContext(ctx, providerConfig))
+	createUpdateRequest := client.NewAuthorizationServerSettings(plan.DefaultScopeDescription.ValueString(), plan.AuthorizationCodeTimeout.ValueInt64(), plan.AuthorizationCodeEntropy.ValueInt64(), plan.RefreshTokenLength.ValueInt64(), plan.RefreshRollingInterval.ValueInt64(), plan.RegisteredAuthorizationPath.ValueString(), plan.PendingAuthorizationTimeout.ValueInt64(), plan.DevicePollingInterval.ValueInt64(), plan.BypassActivationCodeConfirmation.ValueBool())
+	err := addOptionalOauthAuthServerSettingsFields(ctx, createUpdateRequest, plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to add optional properties to add request for OauthAuthServerSettings", err.Error())
+		return
+	}
+	requestJson, err := createUpdateRequest.MarshalJSON()
+	if err == nil {
+		tflog.Debug(ctx, "Update request: "+string(requestJson))
+	}
+	updateOauthAuthServerSettings = updateOauthAuthServerSettings.Body(*createUpdateRequest)
+	updateOauthAuthServerSettingsResponse, httpResp, err := apiClient.OauthAuthServerSettingsApi.UpdateAuthorizationServerSettingsExecute(updateOauthAuthServerSettings)
+	if err != nil {
+		config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while updating OauthAuthServerSettings", err, httpResp)
+		return
+	}
+	// Log response JSON
+	responseJson, err := updateOauthAuthServerSettingsResponse.MarshalJSON()
+	if err == nil {
+		tflog.Debug(ctx, "Read response: "+string(responseJson))
+	}
+	// Read the response
+	readOauthAuthServerSettingsResponse(ctx, updateOauthAuthServerSettingsResponse, &state)
+
+	// Update computed values
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
```

Comment:
I realized these blocks were pointless recently so I removed them from the PD resources (as in the if statement -> return blocks at the very end of methods where they don't actually skip any logic)

Reply:
now removed where applicable